### PR TITLE
Plugin .ini files in icub-main are now automatially generated

### PR DESCRIPTION
**Pull request by randaz81**
pull request on icub-main master as requested

### CODE REVIEW

File: src/libraries/icubmod/imu3DM_GX3/CMakeLists.txt [hunk 1]
```diff
@@ -14,4 +14,11 @@ if (ENABLE_${PROJECTNAME})
     yarp_add_plugin(${PROJECTNAME} 3dm_gx3.cpp 3dm_gx3.h )
     target_link_libraries(${PROJECTNAME} ${YARP_LIBRARIES})
     icub_export_plugin(${PROJECTNAME})
+
+  yarp_install(TARGETS ${PROJECTNAME}
```
Comment (randaz81) [inline src/libraries/icubmod/imu3DM_GX3/CMakeLists.txt]: @barbalberto  can you check this for me? It seems the yarp_install was completely missing...

File: src/libraries/icubmod/canBusSkin/canBusSkin.ini [hunk 2]
```diff
@@ -1,5 +0,0 @@
-[plugin canBusSkin]
-type device
-name canBusSkin
-library canBusSkin
-wrapper skinwrapper
```
Comment (drdanz) [inline src/libraries/icubmod/canBusSkin/canBusSkin.ini]: Warning: the `yarp_prepare_plugin` method uses `grabber`

File: src/libraries/icubmod/canBusVirtualAnalogSensor/canBusVirtualAnalogSensor.ini [hunk 3]
```diff
@@ -1,5 +0,0 @@
-[plugin canBusVirtualAnalogSensor]
-type device
-name canBusVirtualAnalogSensor
-library canBusVirtualAnalogSensor
-wrapper virtualAnalogServer
```
Comment (drdanz) [inline src/libraries/icubmod/canBusVirtualAnalogSensor/canBusVirtualAnalogSensor.ini]: Warning: the `yarp_prepare_plugin` method uses `grabber`

File: src/libraries/icubmod/cartesianController/CMakeLists.txt [hunk 4]
```diff
@@ -37,7 +41,11 @@ if(NOT SKIP_cartesiancontrollerclient)
    target_link_libraries(cartesiancontrollerclient iKin ${YARP_LIBRARIES})
 
    icub_export_plugin(cartesiancontrollerclient)
-   yarp_install(FILES cartesiancontrollerclient.ini DESTINATION ${ICUB_PLUGIN_MANIFESTS_INSTALL_DIR})
+     yarp_install(TARGETS cartesiancontrollerclient
+               COMPONENT Runtime
+               LIBRARY DESTINATION ${ICUB_DYNAMIC_PLUGINS_INSTALL_DIR}
+               ARCHIVE DESTINATION ${ICUB_STATIC_PLUGINS_INSTALL_DIR}
+               YARP_INI DESTINATION ${ICUB_PLUGIN_MANIFESTS_INSTALL_DIR})
```
Comment (drdanz) [inline src/libraries/icubmod/cartesianController/CMakeLists.txt]: Please double check if the generation works properly when there are 2 .ini files generated in the same CMakeLists.txt.

File: src/libraries/icubmod/cfw2Can/cfw2can.ini [hunk 5]
```diff
@@ -1,4 +0,0 @@
-[plugin cfw2can]
-type device
-name cfw2can
-library cfw2can
```
Comment (drdanz) [inline src/libraries/icubmod/cfw2Can/cfw2can.ini]: Please check if `EXTRA_CONFIG WRAPPER=""` is generating a line `wrapper` in the `.ini` file.
I'd remove the `EXTRA_CONFIG` line from CMakeLists.txt

File: src/libraries/icubmod/dragonfly2/CMakeLists.txt [hunk 6]
```diff
@@ -46,7 +46,12 @@ IF(NOT SKIP_dragonfly2)
   yarp_add_plugin(dragonfly2 common/DragonflyDeviceDriver2.h common/DragonflyDeviceDriver2.cpp ${EXTRA_SOURCES})
   target_link_libraries(dragonfly2 ${YARP_LIBRARIES} ${DRAGONFLYAPI_LIB})
   icub_export_plugin(dragonfly2)
-  yarp_install(FILES dragonfly2.ini dragonfly2raw.ini DESTINATION ${ICUB_PLUGIN_MANIFESTS_INSTALL_DIR})
+  
+          yarp_install(TARGETS dragonfly2
+               COMPONENT Runtime
+               LIBRARY DESTINATION ${ICUB_DYNAMIC_PLUGINS_INSTALL_DIR}
+               ARCHIVE DESTINATION ${ICUB_STATIC_PLUGINS_INSTALL_DIR}
+               YARP_INI DESTINATION ${ICUB_PLUGIN_MANIFESTS_INSTALL_DIR})
```
Comment (drdanz) [inline src/libraries/icubmod/dragonfly2/CMakeLists.txt]: Please double check if the 2 .ini files are correct

File: src/libraries/icubmod/embObjAnalog/embObjAnalogSensor.ini [hunk 7]
```diff
@@ -1,5 +0,0 @@
-[plugin embObjAnalogSensor]
-type device
-name embObjAnalogSensor
-library embObjAnalogSensor
-wrapper analogServer
```
Comment (drdanz) [inline src/libraries/icubmod/embObjAnalog/embObjAnalogSensor.ini]: Please double check if it is correct `analogServer` or `AnalogServer`

File: src/libraries/icubmod/embObjFTsensor/embObjFTsensor.ini [hunk 8]
```diff
@@ -1,5 +0,0 @@
-[plugin embObjFTsensor]
-type device
-name embObjFTsensor
-library embObjFTsensor
-wrapper analogServer
```
Comment (drdanz) [inline src/libraries/icubmod/embObjFTsensor/embObjFTsensor.ini]: a - A

File: src/libraries/icubmod/embObjIMU/embObjIMU.ini [hunk 9]
```diff
@@ -1,5 +0,0 @@
-[plugin embObjIMU]
-type device
-name embObjIMU
-library embObjIMU
-wrapper inertial
```
Comment (drdanz) [inline src/libraries/icubmod/embObjIMU/embObjIMU.ini]: inertial vs multipleanalogsensorserver

File: src/libraries/icubmod/embObjInertials/embObjInertials.ini [hunk 10]
```diff
@@ -1,5 +0,0 @@
-[plugin embObjInertials]
-type device
-name embObjInertials
-library embObjInertials
-wrapper analogServer
```
Comment (drdanz) [inline src/libraries/icubmod/embObjInertials/embObjInertials.ini]: a - A

File: src/libraries/icubmod/embObjMais/embObjMais.ini [hunk 11]
```diff
@@ -1,5 +0,0 @@
-[plugin embObjMais]
-type device
-name embObjMais
-library embObjMais
-wrapper analogServer
```
Comment (drdanz) [inline src/libraries/icubmod/embObjMais/embObjMais.ini]: a -A

File: src/libraries/icubmod/embObjMultiEnc/embObjMultiEnc.ini [hunk 12]
```diff
@@ -1,5 +0,0 @@
-[plugin embObjMultiEnc]
-type device
-name embObjMultiEnc
-library embObjMultiEnc
-wrapper analogServer
```
Comment (drdanz) [inline src/libraries/icubmod/embObjMultiEnc/embObjMultiEnc.ini]: a - A

File: src/libraries/icubmod/embObjPSC/embObjPSC.ini [hunk 13]
```diff
@@ -1,5 +0,0 @@
-[plugin embObjPSC]
-type device
-name embObjPSC
-library embObjPSC
-wrapper analogServer
```
Comment (drdanz) [inline src/libraries/icubmod/embObjPSC/embObjPSC.ini]: a - A

File: src/libraries/icubmod/embObjStrain/embObjStrain.ini [hunk 14]
```diff
@@ -1,5 +0,0 @@
-[plugin embObjStrain]
-type device
-name embObjStrain
-library embObjStrain
-wrapper analogServer
```
Comment (drdanz) [inline src/libraries/icubmod/embObjStrain/embObjStrain.ini]: a - A

File: src/libraries/icubmod/esdCan/esdcan.ini [hunk 15]
```diff
@@ -1,5 +0,0 @@
-[plugin ecan]
-type device
-name ecan
-library esdCan
```
Comment (drdanz) [inline src/libraries/icubmod/esdCan/esdcan.ini]: `EXTRA_CONFIG WRAPPER=controlboard`

File: src/libraries/icubmod/esdSniffer/esdSniffer.ini [hunk 16]
```diff
@@ -1,5 +0,0 @@
-[plugin esdSniffer]
-type device
-name esdsniffer
-library esdsniffer
-
```
Comment (drdanz) [inline src/libraries/icubmod/esdSniffer/esdSniffer.ini]: please double check if `EXTRA_CONFIG WRAPPER=""` works

File: src/libraries/icubmod/imuST_M1/imuST_M1.ini [hunk 17]
```diff
@@ -1,5 +0,0 @@
-[plugin imuST_M1]
-type device
-name imuST_M1
-library imuST_M1
-
```
Comment (drdanz) [inline src/libraries/icubmod/imuST_M1/imuST_M1.ini]: double check `EXTRA_CONFIG WRAPPER=inertial`

File: src/libraries/icubmod/plxCan/pcan.ini [hunk 18]
```diff
@@ -1,5 +0,0 @@
-[plugin pcan]
-type device
-name pcan
-library pcan
-
```
Comment (drdanz) [inline src/libraries/icubmod/plxCan/pcan.ini]: Double check `EXTRA_CONFIG WRAPPER=""`

File: src/libraries/icubmod/socketCan/socketcan.ini [hunk 19]
```diff
@@ -1,5 +0,0 @@
-[plugin socketcan]
-type device
-name socketcan
-library socketcan
-
```
Comment (drdanz) [inline src/libraries/icubmod/socketCan/socketcan.ini]: Please double check `EXTRA_CONFIG WRAPPER=controlboard`

File: src/libraries/icubmod/xsensmtx/xsensmtx.ini [hunk 20]
```diff
@@ -1,5 +0,0 @@
-[plugin xsensmtx]
-type device
-name xsensmtx
-library xsensmtx
-
```
Comment (drdanz) [inline src/libraries/icubmod/xsensmtx/xsensmtx.ini]: `EXTRA_CONFIG WRAPPER=inertial`

File: src/libraries/icubmod/imu3DM_GX3/CMakeLists.txt (hunk 1 above)
Comment (barbalberto) [reply src/libraries/icubmod/imu3DM_GX3/CMakeLists.txt]: It was used on Coman/Walkman robots when icubmod was compiled statically.
I think they have imported inthey repositories, anyway this change it is safe, it is fine with me

File: src/libraries/icubmod/cfw2Can/cfw2can.ini (hunk 5 above)
Comment (randaz81) [reply src/libraries/icubmod/cfw2Can/cfw2can.ini]: indeed, it generates a line:
`wrapper ""`

File: src/libraries/icubmod/cartesianController/CMakeLists.txt (hunk 4 above)
Comment (randaz81) [reply src/libraries/icubmod/cartesianController/CMakeLists.txt]: yes, two separated files are are generated correctly

File: src/libraries/icubmod/imu3DM_GX3/CMakeLists.txt (hunk 1 above)
Comment (randaz81) [reply src/libraries/icubmod/imu3DM_GX3/CMakeLists.txt]: 👍

File: src/libraries/icubmod/embObjMais/embObjMais.ini (hunk 11 above)
Comment (randaz81) [reply src/libraries/icubmod/embObjMais/embObjMais.ini]: fixed

File: src/libraries/icubmod/embObjMultiEnc/embObjMultiEnc.ini (hunk 12 above)
Comment (randaz81) [reply src/libraries/icubmod/embObjMultiEnc/embObjMultiEnc.ini]: fixed

File: src/libraries/icubmod/embObjPSC/embObjPSC.ini (hunk 13 above)
Comment (randaz81) [reply src/libraries/icubmod/embObjPSC/embObjPSC.ini]: fixed

File: src/libraries/icubmod/embObjAnalog/embObjAnalogSensor.ini (hunk 7 above)
Comment (randaz81) [reply src/libraries/icubmod/embObjAnalog/embObjAnalogSensor.ini]: fixed

File: src/libraries/icubmod/embObjFTsensor/embObjFTsensor.ini (hunk 8 above)
Comment (randaz81) [reply src/libraries/icubmod/embObjFTsensor/embObjFTsensor.ini]: fixed

File: src/libraries/icubmod/embObjInertials/embObjInertials.ini (hunk 10 above)
Comment (randaz81) [reply src/libraries/icubmod/embObjInertials/embObjInertials.ini]: fixed

File: src/libraries/icubmod/esdCan/esdcan.ini (hunk 15 above)
Comment (randaz81) [reply src/libraries/icubmod/esdCan/esdcan.ini]: fixed

File: src/libraries/icubmod/plxCan/pcan.ini (hunk 18 above)
Comment (randaz81) [reply src/libraries/icubmod/plxCan/pcan.ini]: fixed (removed)

File: src/libraries/icubmod/socketCan/socketcan.ini (hunk 19 above)
Comment (randaz81) [reply src/libraries/icubmod/socketCan/socketcan.ini]: fixed (removed)

File: src/libraries/icubmod/esdSniffer/esdSniffer.ini (hunk 16 above)
Comment (randaz81) [reply src/libraries/icubmod/esdSniffer/esdSniffer.ini]: fixed (removed)

File: src/libraries/icubmod/cfw2Can/cfw2can.ini (hunk 5 above)
Comment (randaz81) [reply src/libraries/icubmod/cfw2Can/cfw2can.ini]: fixed (removed)

File: src/libraries/icubmod/dragonfly2/CMakeLists.txt (hunk 6 above)
Comment (randaz81) [reply src/libraries/icubmod/dragonfly2/CMakeLists.txt]: yes, they are correct.

File: src/libraries/icubmod/canBusSkin/canBusSkin.ini (hunk 2 above)
Comment (randaz81) [reply src/libraries/icubmod/canBusSkin/canBusSkin.ini]: fixed to skinwrapper

File: src/libraries/icubmod/canBusVirtualAnalogSensor/canBusVirtualAnalogSensor.ini (hunk 3 above)
Comment (randaz81) [reply src/libraries/icubmod/canBusVirtualAnalogSensor/canBusVirtualAnalogSensor.ini]: fixed to virtualAnalogServer

File: src/libraries/icubmod/embObjIMU/embObjIMU.ini (hunk 9 above)
Comment (randaz81) [reply src/libraries/icubmod/embObjIMU/embObjIMU.ini]: fixed to inertial

File: src/libraries/icubmod/imuST_M1/imuST_M1.ini (hunk 17 above)
Comment (randaz81) [reply src/libraries/icubmod/imuST_M1/imuST_M1.ini]: correct

File: src/libraries/icubmod/xsensmtx/xsensmtx.ini (hunk 20 above)
Comment (randaz81) [reply src/libraries/icubmod/xsensmtx/xsensmtx.ini]: correct

File: src/libraries/icubmod/bcbBattery/bcbBattery.ini [hunk 21]
```diff
@@ -1,5 +0,0 @@
-[plugin bcbBattery]
-type device
-name bcbBattery
-library bcbBattery
-wrapper batteryWrapper
```
Comment (drdanz) [inline src/libraries/icubmod/bcbBattery/bcbBattery.ini]: I think we missed this one on the first review, or perhaps it was wrong before...
Anyway `wrapper batteryWrapper` -> `EXTRA_CONFIG WRAPPER=BatteryWrapper`

File: src/libraries/icubmod/bmsBattery/bmsBattery.ini [hunk 22]
```diff
@@ -1,5 +0,0 @@
-[plugin bmsBattery]
-type device
-name bmsBattery
-library bmsBattery
-wrapper batteryWrapper
```
Comment (drdanz) [inline src/libraries/icubmod/bmsBattery/bmsBattery.ini]: B -> b

File: src/libraries/icubmod/bcbBattery/bcbBattery.ini (hunk 21 above)
Comment (randaz81) [reply src/libraries/icubmod/bcbBattery/bcbBattery.ini]: fixed by 9e0dee0ef0753f6536f6c636a43913

File: src/libraries/icubmod/bmsBattery/bmsBattery.ini (hunk 22 above)
Comment (randaz81) [reply src/libraries/icubmod/bmsBattery/bmsBattery.ini]: fixed by 9e0dee0ef0753f6536f6c636a43913

**Comment by randaz81** [comment]
Ready to be merged @drdanz

**Pull request merged**